### PR TITLE
Move to Elastic.Transport nuget package

### PR DESCRIPTION
**Pull request by Mpdreamz**
This moves our `master` branch to `Elastic.Transport` which now lives over at:

https://github.com/elastic/elastic-transport-net

### CODE REVIEW

File: src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs [hunk 1]
```diff
@@ -54,8 +47,9 @@ public ConnectionConfiguration(IConnectionPool connectionPool, IConnection conne
 	public abstract class ConnectionConfigurationBase<TConnectionConfiguration> : TransportConfigurationBase<TConnectionConfiguration>, IConnectionConfigurationValues
 		where TConnectionConfiguration : ConnectionConfigurationBase<TConnectionConfiguration>, IConnectionConfigurationValues
 	{
-		protected ConnectionConfigurationBase(IConnectionPool connectionPool, IConnection connection, ITransportSerializer serializer)
-			: base(connectionPool, connection, serializer, ElasticsearchProductRegistration.Default) =>
+		protected ConnectionConfigurationBase(IConnectionPool connectionPool, IConnection connection, ITransportSerializer serializer,
+			IProductRegistration registration = null)
```
Comment (stevejgordon) [inline src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs:51]: Is there a case we imagine where someone would want to pass in a different `IProductRegistration` within the client context?

File: src/Nest/Cluster/TaskManagement/CancelTasks/CancelTasksResponse.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Collections.Generic;
 using System.Runtime.Serialization;
 using Elastic.Transport;
+using Elastic.Transport.Products.Elasticsearch.Failures;
```
Comment (stevejgordon) [inline src/Nest/Cluster/TaskManagement/CancelTasks/CancelTasksResponse.cs:8]: Is this namespace a bit verbose? Could we loose "".Failures" in transport?

File: src/Nest/CommonAbstractions/NestElasticsearchProductConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using Elastic.Transport;
+using Elastic.Transport.Products;
+using Elastic.Transport.Products.Elasticsearch;
+
+namespace Nest
+{
+	public class NestElasticsearchProductConfiguration : ElasticsearchProductRegistration
```
Comment (stevejgordon) [inline src/Nest/CommonAbstractions/NestElasticsearchProductConfiguration.cs:11]: Perhaps rename?

```suggestion
	public class NestElasticsearchProductRegistration: ElasticsearchProductRegistration
```

**Review by stevejgordon — APPROVED** [review]
Not as scary as it looked from the change count!! LGTM! A couple of questions/minor optional suggestions. Nothing blocking this merging as-is though.

File: src/Nest/Cluster/TaskManagement/CancelTasks/CancelTasksResponse.cs (hunk 2 above)
Comment (Mpdreamz) [reply src/Nest/Cluster/TaskManagement/CancelTasks/CancelTasksResponse.cs:8]: Good callout opened: https://github.com/elastic/elastic-transport-net/issues/21 to track and discuss :)

File: src/Nest/CommonAbstractions/NestElasticsearchProductConfiguration.cs (hunk 3 above)
Comment (Mpdreamz) [reply src/Nest/CommonAbstractions/NestElasticsearchProductConfiguration.cs]: :+1: ! amended the PR

File: src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs (hunk 1 above)
Comment (Mpdreamz) [reply src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs:51]: aye `NEST` uses a slightly different registration then the low level client :)

**Pull request merged**